ww4ff/run/train.py

```diff
@@ -6,40 +6,44 @@
 from torch.optim.adamw import AdamW
 import torch
 import torch.nn as nn
-import torch.utils.data as tud
 
 from .args import ArgumentParserBuilder, opt
 from .preprocess_dataset import print_stats
 from ww4ff.data.dataset import FlatWavDatasetLoader, WakeWordTrainingDataset, WakeWordEvaluationDataset
 from ww4ff.data.dataloader import StandardAudioDataLoaderBuilder
 from ww4ff.data.transform import compose, ZmuvTransform, StandardAudioTransform, batchify, random_slice
 from ww4ff.settings import SETTINGS
-from ww4ff.model import find_model, model_names
+from ww4ff.model import find_model, model_names, InferenceEngine
 from ww4ff.utils.random import set_seed
 
 
 def main():
-    def evaluate(data_loader, prefix: str):
+    def evaluate(dataset: WakeWordEvaluationDataset, prefix: str):
         model.eval()
-        pbar = tqdm(data_loader, position=1, desc=prefix, leave=True)
+        ds_iter = iter(dataset)
         c = Counter()
-        for batch in pbar:
-            batch.to(device)
-            scores = model(zmuv_transform(std_transform(batch.audio_data)),
-                           std_transform.compute_lengths(batch.lengths))
-            preds = scores.max(1)[1].view(-1).tolist()
-            for pred, label in zip(preds, batch.labels.tolist()):
-                if pred == 1 and label == 1:
+        engine = InferenceEngine(model, zmuv_transform)
+        last_idx = None
+        with tqdm(position=1, desc=prefix, leave=True) as pbar:
+            while True:
+                try:
+                    example = next(ds_iter)
+                except StopIteration:
+                    break
+                if last_idx != ds_iter.curr_file_idx:
+                    engine.reset()
+                    last_idx = ds_iter.curr_file_idx
```

Comment:
why do you need to keep track of `last_idx`?

Reply:
The iterable dataset emits segments of audio across one file. When we reach a new file, we reset the engine.